### PR TITLE
Jsdoc types

### PR DESCRIPTION
**Pull request by natjoe4**
# Summary
Added JSDoc types and improved comments. Also removed the `res.status = () => res` workaround added back when we were using Asymmetrik

## New behavior
Hopefully none

## Code changes
Aside from updating comments, simply removing the above-mentioned workaround.

# Testing guidance

- Give it an `npm run test` and check the $import functionality to ensure that nothing has changed.
- Look through each file in the test server and make sure all appropriate functions have JSDocs and all JSDocs have correct argument typing

### CODE REVIEW

**Review by lmd59 — COMMENTED** [review]
Do we want types on the `@returns` as well?

File: src/services/bundle.service.js [hunk 1]
```diff
@@ -11,10 +11,10 @@ const logger = loggers.get('default');
 
 /**
  * Creates transaction-response Bundle
- * @param {*} results - request results
- * @param {*} res - an object containing the response
- * @param {*} type - bundle type
- * @param { boolean } xprovenanceIncluded - X-Provenance header was included and
+ * @param {Array} results - an array of equest result objects
```
Comment (lmd59) [inline src/services/bundle.service.js:14]:
```suggestion
 * @param {Array} results - an array of request result objects
```

Comment (lmd59) [reply src/services/bundle.service.js:14]: I also noticed the `-` in the jsdoc comments here. Not in scope, but do we want to establish an official style for this?

File: src/services/measure.service.js [hunk 2]
```diff
@@ -90,9 +90,9 @@ const search = async (args, { req }) => {
  * takes a measureReport and a set of required data with which to calculate the measure and
```
Comment (lmd59) [inline src/services/measure.service.js:90]:
```suggestion
 * Takes a measureReport and a set of required data as part of the request. Calculates the measure and
```

File: src/services/measure.service.js [hunk 3]
```diff
@@ -90,9 +90,9 @@ const search = async (args, { req }) => {
...
  * creates new documents for the measureReport and requirements in the appropriate collections.
```
Comment (lmd59) [inline src/services/measure.service.js:91]:
```suggestion
 * creates new documents for the measureReport and required data in the appropriate collections.
```

Comment (lmd59) [reply src/services/measure.service.js:91]: These aren't on you, but trying to read this hurt my brain a little, so I changed it. Feel free to reject if you disagree

File: src/services/measure.service.js [hunk 4]
```diff
@@ -166,8 +166,8 @@ const submitData = async (args, { req }) => {
  * Retrieves measure bundle from the measure ID and
  * maps data requirements into an export request, which is
  * returned to the intiial import client.
```
Comment (lmd59) [inline src/services/measure.service.js:168]:
```suggestion
 * returned to the initial import client.
```

File: src/util/bundleUtils.js [hunk 5]
```diff
@@ -7,6 +7,13 @@ const supportedResources = require('../server/supportedResources');
 // lookup from patient compartment-definition
 const patientRefs = require('../compartment-definition/patient-references');
 
+/**
+ * Converts an array of FHIR resources to a fhir searchset bundle
```
Comment (lmd59) [inline src/util/bundleUtils.js:11]: It's silly. I know

File: src/util/bundleUtils.js [hunk 6]
```diff
@@ -7,6 +7,13 @@ const supportedResources = require('../server/supportedResources');
...
+ * @param {Array} resources and array of FHIR resources
```
Comment (lmd59) [inline src/util/bundleUtils.js:12]:
```suggestion
 * @param {Array} resources an array of FHIR resources
```

File: src/services/bundle.service.js (hunk 1 above)
Comment (natjoe4) [reply src/services/bundle.service.js]: I like the dashes!

File: src/util/bundleUtils.js [hunk 7]
```diff
@@ -138,7 +145,7 @@ async function assembleCollectionBundleFromMeasure(measure) {
 /**
  * Go through the relatedArtifact ValueSets and query for them from the database
  * @param {Object} lib FHIR library to grab ValueSets for
- * @returns list of ValueSet resources required by the library
+ * @returns {Array} list of ValueSet resources required by the library
```
Comment (lmd59) [inline src/util/bundleUtils.js:148]:
```suggestion
 * @returns {Array} array of ValueSet resources required by the library
```

**Review by lmd59 — APPROVED** [review]

**Pull request merged**